scripts/survival_migration/survivalDataMigration.py

```diff
@@ -0,0 +1,196 @@
+    newVocAddingToZero = []
+    newVocAddingToOne = []
+    # just defined one or zero
+    for splitString in zeroAndOneSplit:
+        mapAndValuesSplit = splitString.split(':')
+        if (len(mapAndValuesSplit) == 2):
+            mapId = mapAndValuesSplit[0]
+            values = mapAndValuesSplit[1].split(',')
+            if mapId == '0':
+                for value in values:
+                    if value.lower() not in zeroVoc:
+                        zeroVoc.append(value.lower())
+                        newVocAddingToZero.append(value.lower())
+            elif mapId == '1':
+                for value in values:
+                    if value.lower() not in oneVoc:
+                        oneVoc.append(value.lower())
+                        newVocAddingToOne.append(value.lower())
+        else:
+            print("additionalVocabularies format is incorrect, please check the example by using -h option")
+
+    # save vocabularies to file
+    separator = ','
+    oneUpdateString = separator.join(newVocAddingToOne)
+    zeroUpdateString = separator.join(newVocAddingToZero)
+    df = pd.read_csv(vocabulariesFile, sep='\t')
+
+    if len(newVocAddingToZero) > 0 or len(newVocAddingToOne) > 0:
+        # get vocabularies
+        df2 = pd.DataFrame([[oneUpdateString, zeroUpdateString]], columns=[MAPPING_TO_ONE_COLUMN, MAPPING_TO_ZERO_COLUMN])
+        df = df.append(df2, ignore_index=True)
+        if len(newVocAddingToOne) > 0:
+            print("vocabulary {} has been added to one mapping".format(oneUpdateString))
+        if len(newVocAddingToZero) > 0:
+            print("vocabulary {} has been added to zero mapping".format(zeroUpdateString))
+
+    df.to_csv(vocabulariesFile, index=False, sep='\t', header=True)
+
+def migrate_file(args):
+    clinicalFile = args.clinicalFile
+    vocabulariesFile = args.vocabulariesFile
+    shouldOverrive = args.override
+    newFile = os.path.dirname(args.clinicalFile) + "/new_data_clinical_patients.txt"
+    tmpFile = os.path.dirname(args.clinicalFile) + "/tmp_data_clinical_patients.txt"
+
+    # get vocabularies
+    vocabulariesMappingToOne = []
+    vocabulariesMappingToZero = []
+    generateVocabularies(vocabulariesMappingToOne, vocabulariesMappingToZero, vocabulariesFile)
+
+    if args.additionalVocabularies != None and args.additionalVocabularies != '':
+        splitAdditionalVocabularies(args.additionalVocabularies, vocabulariesMappingToOne, vocabulariesMappingToZero, vocabulariesFile)
+
+    print("vocabularies will be mapping to one: ")
+    print(vocabulariesMappingToOne)
+    print("vocabularies will be mapping to zero: ")
+    print(vocabulariesMappingToZero)
+
+    # write files
+    input = open(clinicalFile, 'rb')
+    output = open(newFile, 'wb')
+    tmp_output = open(tmpFile, 'wb')
+
+    writer = csv.writer(output)
+    tmpwritter = csv.writer(tmp_output)
+    for row in csv.reader(input):
+        if row[0].startswith('#'):
+            writer.writerow(row)
```

Comment:
If the script terminates because it couldn't find a value in vocabulary, the intermediate data and tmp file created exists. Should they be deleted? I think its better to save comments to a variable here and write to file at the end?

Reply:
Since we also want to remove the tmp file and the result file if program exit, should we log this error in console instead of the result file?